JavaScript/JavaScript_Limitations.md

```diff
@@ -0,0 +1,85 @@
+````
+For example, assume a script in the document at `http://shop.company.com/dir/other.html`
+which executes the following statement:
+
+document.domain = "company.com";
+
+After execution, the page would pass the origin check with http://company.com/dir/page.html.
+However, by the same reasoning, company.com could not set document.domain to othercompany.com.
+
+````
+Using this you can execute javascript from an iframe.  Please note that this is not suitable for Firefox which will not allow a change to `document.domain`
+
+Another notable method is `window.postMessage` which safely enables cross-origin communication.  Cross-window messaging security model is two-sided. The sender ensures that the receiving domain is targetDomain. The receiver checks that the message came from proper event.origin.
+
+````
+postMessage(data, targetDomain), where:
+
+data - The message. According to the specification, it could be any object. But as of now, only strings are supported in major browsers.
+
+targetDomain - Limit receiving iframe by given domain. Can contain ‘*’ which doesn’t put any restrictions.
+````
+
+
+#### Database Access
+
+Accessing databases, in particular backend databases. This is a problem without the help of a server side script like Ajax which performs the db access.  
+
+For example if a user was to select an item in a dropdown list, which is implemented by JavaScript and these items are in a database.  A function could be used to show the dropdown details.  This function would do the following.
+1. chick if item is selected.
+2. Create a XMLHttpRequest object
+3. Create the function to be executed when the server response is ready
+4. Send the request off to a file on the server.
+
+Then in turn the page on the server called by the JavaScript above is an ASP file called "getdetails.asp".  This file would run a query against a database, and returns the result in an HTML table
+
+
+#### Writing files to the server
+
+Writing files to the server.  JavaScript cannot write to files on the server without the help of a server side script. Using Ajax, JavaScript can send a request to the server, which can read a file in XML or plain text format but it cannot write to a file unless the file called on the server actually runs as a script to do the file write for you.
+
+````
+$.ajax({
+  type: 'POST',
+  url: url,//url of receiver file on server
+  data: data, //your data
+  success: success, //callback when ajax request finishes
+  dataType: dataType //text/json...
+});
+
+````
+
+#### Other
+
+Other limitations of note are that JavaScript cannot close a window if it didn't open it, an obvious security reason.  Also protection of page source or images, this is not done by JavaScript.  Finally the option to disable JavaScript, many users disable it for security reasons.  If your site is highly scripted and the user has disabled JavaScript then it can be disadvantageous to both parties.
```

Comment:
Should this be 'JavaScript cannot close a window if it didn't open it, for obvious security reasons.'?